lib/trino-parquet/src/main/java/io/trino/parquet/reader/TimeMillisColumnReader.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader;
+
+import io.trino.parquet.PrimitiveField;
+import io.trino.spi.TrinoException;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.type.TimeType;
+import io.trino.spi.type.Timestamps;
+import io.trino.spi.type.Type;
+
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static java.lang.String.format;
+
```

Comment:
"Support Time(3) type in the old Parquet column readers" doesn't modify any tests

- is it test-covered?
- do we have test coverage ensuring both batched and "old" readers return same values?
- do we have compatibility tests for Trino vs Hive? Hive is our reference implementation for Hive type handling.
   - product test coverage ensures we're not accidentally biased toward a particular Hive version

Reply:
AFAIK Hive does not support TIME(3) type, and out "old reader" tests are base almost purely on hive. So comparing the old reader to the new one on types not supported by hive needs writing additional testing framework.

Reply:
AFAIK Hive, Iceberg and Delta don't support time(3). This is more for adherence with parquet spec and preventing exception/incorrect results in custom connectors or in future if some trino connector adds support for time(3) on parquet files. @skrzypo987 could you take a parquet file generated from the tool where you saw this encoding, add that to test resources and read it in a test in trino-parquet ?